javalib/src/main/scala/java/util/SplittableRandom.scala

```diff
@@ -0,0 +1,122 @@
+package java.util
+
+/*
+ * This is a clean room implementation derived from the original paper
+ * and Java implementation mentioned there:
+ *
+ * Fast Splittable Pseudorandom Number Generators
+ * by Guy L. Steele Jr., Doug Lea, Christine H. Flood
+ * http://gee.cs.oswego.edu/dl/papers/oopsla14.pdf
+ *
+ */
+private object SplittableRandom {
+
+  private final val DoubleULP = 1.0 / (1L << 53)
+  private final val GoldenGamma = 0x9e3779b97f4a7c15L // odd
+
+  private var defaultGen: Long = new Random().nextLong()
+
+  private def nextDefaultGen(): Long = {
+    val s = defaultGen
+    defaultGen = s + (2 * GoldenGamma)
+    s
+  }
+
+  case class GenLong(value: Long)
+
+  private final def mix64ForGamma(z: Long): Long = {
+    val z1 = (z ^ (z >>> 33)) * 0xff51afd7ed558ccdL
+    val z2 = (z1 ^ (z1 >>> 33)) * 0xc4ceb9fe1a85ec53L
+    z2 ^ (z2 >>> 33)
+  }
+
+  /*
+   * This function implements David Stafford's variant 4,
+   * while the paper version uses the original MurmurHash3 finalizer
+   * reference:
+   * http://zimbry.blogspot.pt/2011/09/better-bit-mixing-improving-on.html
+   */
+  private final def mix32(z: Long): Int = {
+    val z1 = (z ^ (z >>> 33)) * 0x62a9d9ed799705f5L
+    val z2 = (z1 ^ (z1 >>> 28)) * 0xcb24d0a5c88c35b3L
+    (z2 >>> 32).toInt
+  }
+
+  /*
+   * function implements Stafford's variant 13,
```

Comment:
I think this comment is missing a "This" at the beginning of the sentence, if I compare it to the one for `mix32`.